app/Http/Controllers/StatusController.php

```diff
@@ -19,14 +19,27 @@
 
 class StatusController extends Controller
 {
-    public static function getStatus($statusId)
-    {
-        return Status::where('id', $statusId)->with('user',
-                                                    'trainCheckin',
-                                                    'trainCheckin.Origin',
-                                                    'trainCheckin.Destination',
-                                                    'trainCheckin.HafasTrip',
-                                                    'event')->withCount('likes')->firstOrFail();
+    public static function getStatus($statusId) {
+        $status = Status::where('id', $statusId)->with('user',
+                                                       'trainCheckin',
+                                                       'trainCheckin.Origin',
+                                                       'trainCheckin.Destination',
+                                                       'trainCheckin.HafasTrip',
+                                                       'event')->withCount('likes')->firstOrFail();
+
+        $authID = null;
+
+        //PrivateProfile change to "also following"
+        if (Auth::check()) {
+            $authID = Auth::user()->id;
+        }
+
+        if ($status->user->id == $authID || !$status->user->private_profile) {
+            return $status;
+        }
+
+        abort(403);
```

Comment:
```suggestion
        //PrivateProfile change to "also following"
        if($status->user->private_profile && (!Auth::check() || Auth::user()->id != $status->user->id)) {
           abort(403);
        }

        return $status;
```

Reply:
This will not work since `Auth::user()` has no attribute `id` if a profile page visitor is not logged in. Thats why the variable `authID` is needed with a null initialization.

Reply:
But thats why ```!Auth::check()``` is checked before?
If user isn't logged in (!Auth::check() is true) then the second condition doesn't get checked.

Reply:
This will only work in an AND-case. OR doesn't work that way. If `Auth::check` equals false, `Auth::user()->id != $status->user->id` might still return true which is why this method always will be accessed.

Reply:
If the User isn't logged in the first condition will return ```true```, so the second condition doesn't get checked.
It the User is logged in, the first condition is ```false```and the ```Auth::user()->id``` is called.

See: 
<img width="838" alt="Screen Shot 2020-12-02 at 13 18 18" src="https://user-images.githubusercontent.com/4103693/100871632-dfdac400-34a0-11eb-8cb8-925c9f2db15e.png">
Source: https://www.php.net/manual/de/language.operators.logical.php

Reply:
Ah. Oh. Yeah, I'm sorry. This makes sense. For some reason I thought, `!Auth::check()` equals to false, not true... smh :man_facepalming:

Reply:
![giphy](https://user-images.githubusercontent.com/4103693/100872172-ae162d00-34a1-11eb-853f-17718550cba5.gif)